docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+- hello.abi, the application binary interface (ABI) file for the smart contract.
+
+#### Deploy the hello Smart Contract (built with cdt-cpp)
+
+Execute the following command to deploy the `hello` smart contract to the account `hello` which you created at the beginning:
+
+```shell
+dune --deploy <PATH_TO_HELLO_DIR>
+```
+
+### Create, Build and Deploy a DUNE cmake Application
+
+The DUNE cmake application is a more complex type than the bare type and is suited better for larger projects which involve multiple contracts and/or web applications.
+
+```shell
+dune --create-cmake-app hello .
+cd hello
+ls
+```
+
+The result of this command is a `hello` directory with a bit more complex structure than the bare application because it generates some extra files to configure the cmake project:
+
+- CMakeLists.txt, cmake configuration file.
+- README.txt, a text file which contains information about how to build this application with cmake.
+- build, the output build folder, at the beginning is empty.
+- include, the C++ include files folder, at the beginning it contains just hello.hpp file.
+- ricardian, the folder contains the smart contract ricardian definition, hello.contracts.md file.
+- src, the C++ implementation files folder, at the beginning it contains just hello.cpp file.
+
+#### Build the DUNE cmake Application
+
+To build the DUNE cmake application run the following command:
+
+```shell
+dune --cmake-build <PATH_TO_CMakeLists.txt_PARENT_DIR>
+ls <PATH_TO_CMakeLists.txt_PARENT_DIR>/build/hello
+```
+
+You can run the two commands from above from any directory on your local machine, just make sure you correctly replace the `<PATH_TO_CMakeLists.txt_PARENT_DIR>` with the parent directory path of `CMakeLists.txt` file.
+
+The result of the above build command are two files located in the `./build/hello/` folder:
+
+- hello.wasm, the WebAssembly binary file for the smart contract.
+- hello.abi, the application binary interface (ABI) file for the smart contract.
+
+#### Deploy the hello Smart Contract (built with cmake)
+
+Execute the following command to deploy the `hello` smart contract to the account `hello` which you created at the beginning:
+
+```shell
+dune --deploy <PATH_TO_CMakeLists.txt_PARENT_DIR>/build/hello
+```
+
+## First Test
+
+Let's send the `hi` action to the local node and send as input parameter the test account name you created earlier:
+
+```shell
+dune --send-action hello hi '[ama]' hello@active
+```
+
+The output of the command above shows on the first line that `hello::hi` action was executed with the input parameter `{"nm":"ama"}` and on the second line the output of the action itself `Name: ama`.
+
+```txt
+#         hello <= hello::hi                    {"nm":"ama"}
+>> Name: ama
+```
+
+## Smart Contract Source Files
+
+The smart contract C++ source files are:
+
+- hello.hpp
+- hello.cpp
+
+### The hello.hpp File
+
+The `hello.hpp` is the C++ header file and it contains the `hello` smart contract C++ class definition.
```

Comment:
Pretty likely that the devs don't know what a header file is or why it's useful. 
Need to give a brief explanation about it.

Reply:
I thought this guide assumes the user knows basics of C++. Do we really wanna go into the C++ explanation of what h/hpp and cpp files are?

Reply:
If we go the direction of assuming they know c++ then we are cutting down the available dev pool significantly.

We should aim to give them all the info they need to get started as if the language is new to them, but not programming being new to them.

Reply:
I added more about hpp and cpp files